src/core/namespace.c

```diff
@@ -1090,12 +1104,17 @@ static int mount_procfs(const MountEntry *m, const NamespaceInfo *ns_info) {
                  * per-instance, we'll exclusively use the textual value for hidepid=, since support was
                  * added in the same commit: if it's supported it is thus also per-instance. */
 
-                opts = strjoin("hidepid=",
-                               ns_info->protect_proc == PROTECT_PROC_DEFAULT ? "off" :
-                               protect_proc_to_string(ns_info->protect_proc),
-                               ns_info->proc_subset == PROC_SUBSET_PID ? ",subset=pid" : "");
-                if (!opts)
-                        return -ENOMEM;
+                const char *hpv = ns_info->protect_proc == PROTECT_PROC_DEFAULT ?
+                                "off" :
+                                protect_proc_to_string(ns_info->protect_proc);
+
+                if (mount_option_supported("proc", "hidepid", hpv))
+                        if (!strextend(&opts, "hidepid=", hpv))
```

Comment:
strextend() is a bit misleading here. you rely on the fact that opts is so far NULL (because otherwise you'd have to use strextend_with_separator() to add in the ","). But you use strextend() instead of strjoin(), which suggests you actually want to extend something that was already there...